src/_guides/libraries/testing.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "Dart Testing"
+description: "How to test Dart, Flutter, and 'Dart for the Web' applications?"
+---
+
+Software testing, an important part of app development, helps verify that
+your app is working correctly before unleashing it on your users.
+This Dart testing guide outlines several types of testing, and points
+you to where you can learn more about testing on the Flutter, Dart VM,
+and 'Dart for the Web' platforms.
+
+## Testing types
+
+The Dart testing docs focus on three types of testing, out of the
+[many testing types](http://www.aptest.com/testtypes.html) that
+you may be familiar with: unit, widget (or component, depending on the
+platform), and integration (or end-to-end). Testing terminology varies,
+but these are the terms and concepts that you are likely to
+encounter when using Dart technologies:
+
+* _Unit_ tests focus on verifying the smallest piece of testable
+  software, such as a function, method, or class. Your testing suite
+  should have more unit tests that widget or integration tests.
+
+* _Widget_ (or _component_) tests verify that a widget (which
+  usually consists of multiple classes) behaves as expected.
+  A widget test requires a mock framework that can mimic user actions,
+  events, perform layout, and instantiate child widgets.
+  Note that Flutter refers to widgets, and AngularDart refers to
+  components.
+
+* _Integration_ or (_end-to-end_) tests verify the behavior of
```

Comment:
At a minimum this should be fixed to put the "or" inside the parentheses. But I might be more in favor of writing:
> * _Integration_ and _end-to-end_ tests ...

given that E2E is a form of integration testing whose terminology is used mainly for web apps.